hoomd/hpmc/tune/move_size.py

```diff
@@ -0,0 +1,390 @@
+                currently ignored.
+        """
+        if self.is_attached:
+            # update maximum move sizes
+            if self._update_move_sizes:
+                for tunable in self._tunables:
+                    if tunable.attr == 'a':
+                        max_move_size = self.max_rotation_move[tunable.type]
+                    else:
+                        max_move_size = self.max_translation_move[tunable.type]
+                    tunable.domain = (self._min_move_size, max_move_size)
+
+            tuned = self.solver.solve(self._tunables)
+            self._tuned = self._tuned + 1 if tuned else 0
+
+    def _update_tunables(self, *, new_moves=tuple(), new_types=tuple()):
+        tunables = self._tunables
+
+        # First filter out any move size tune definitions that don't match
+        # the new specification.
+        def filter_tunables(tunable):
+            return ((new_moves is None or tunable.attr in new_moves) and
+                    (new_types is None or tunable.type in new_types))
+
+        self._tunables = list(filter(filter_tunables, tunables))
+        tune_definitions = set(self._tunables)
+
+        # Add any move size tune definitions that are required by the new
+        # specification.
+        for move in new_moves:
+            for new_type in new_types:
+                if move == 'a':
+                    max_move_size = self.max_rotation_move[new_type]
+                else:
+                    max_move_size = self.max_translation_move[new_type]
+                move_definition = _MoveSizeTuneDefinition(
+                    move, new_type, self.target,
+                    (self._min_move_size, max_move_size))
+                if move_definition not in tune_definitions:
+                    self._tunables.append(move_definition)
+
+    def _update_tunables_attr(self, attr, value):
+        for tunable in self._tunables:
+            setattr(tunable, attr, value)
+
+
+class MoveSize(_InternalCustomTuner):
+    """Tunes HPMCIntegrator move sizes to targeted acceptance rate.
+
+    For most common creation of a `MoveSize` tuner see `MoveSize.secant_solver`
+    and `MoveSize.scale_solver` respectively.
+
+    Args:
+        trigger (hoomd.trigger.Trigger): ``Trigger`` to determine when to run
+            the tuner.
+        moves (list[str]): A list of types of moves to tune. Available options
+            are 'a' and 'd'.
+        target (float): The acceptance rate for trial moves that is desired. The
+            value should be between 0 and 1.
+        solver (`hoomd.tune.SolverStep`): A solver that tunes move sizes to
+            reach the specified target.
+        types (list[str]): A list of string particle types to tune the move
+            size for, defaults to None which upon attaching will tune all types
+            in the system currently.
+        max_translation_move (float): The maximum value of a translational move
+            size to attempt.
```

Comment:
For the `max_*_move` values, I assume that `None` implies no maximum. Document this.